arcgis-ios-sdk-samples/Maps/Display layer view state/LayerStatusViewController.swift

```diff
@@ -70,43 +66,42 @@ class LayerStatusViewController: UIViewController, UITableViewDataSource, UITabl
         //assign a closure for layerViewStateChangedHandler, in order to receive layer view status changes
         self.mapView.layerViewStateChangedHandler = { [weak self] (layer: AGSLayer, state: AGSLayerViewState) in
             DispatchQueue.main.async {
-                guard let strongSelf = self else { return }
+                guard let self = self else { return }
                 //find the index of layer in operational layers list
                 //and update its status
-                let index = strongSelf.map.operationalLayers.index(of: layer)
+                let index = self.map.operationalLayers.index(of: layer)
                 if index != NSNotFound {
-                    strongSelf.viewStatusArray[index] = strongSelf.viewStatusString(state.status)
-                    
-                    strongSelf.tableView.reloadData()
+                    let indexPath = IndexPath(row: index, section: 0)
+                    self.tableView.reloadRows(at: [indexPath], with: .none)
                 }
             }
         }
         
         //setup source code bar button item
         (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["LayerStatusViewController"]
     }
-    
-    //initialize status array to `Unknown`
-    func populateViewStatusArray() {
-        for _ in 0..<map.operationalLayers.count {
-            self.viewStatusArray.append("Unknown")
-        }
-    }
 
     //return string for current status name
     func viewStatusString(_ status: AGSLayerViewStatus) -> String {
-        switch status {
-        case AGSLayerViewStatus.active:
-            return "Active"
-        case AGSLayerViewStatus.notVisible:
-            return "Not Visible"
-        case AGSLayerViewStatus.outOfScale:
-            return "Out of Scale"
-        case AGSLayerViewStatus.loading:
-            return "Loading"
-        case AGSLayerViewStatus.error:
-            return "Error"
-        default:
+        var statuses = [String]()
+        if status.contains(.active) {
+            statuses.append("Active")
```

Comment:
@philium I got a couple of questions
- Since the status at a time can only be one of these, can `if... else` be used? In that case, is there an advantage of the if statement over the switch case?
- Do we need to handle for the same status getting appended twice?

Reply:
> the status at a time can only be one of these

Since `AGSLayerViewStatus` is an option set (instead of an enum), it can have more than one status at a time. Therefore these do need to be separate `if` statements so that all of the statuses in the set are shown to the user.

When running the sample, it does briefly show multiple statuses.

> Do we need to handle for the same status getting appended twice?

No. Each of the statuses (`active`, `notVisible`, etc.) should be unique, so `statuses` should never contain duplicates.

Reply:
Ok good to know. Thanks